src/java.base/share/classes/java/lang/ref/Reference.java

```diff
@@ -325,14 +325,42 @@ public void runFinalization() {
      * been cleared, either by the program or by the garbage collector, then
      * this method returns {@code null}.
      *
+     * @apiNote
+     * This method returns a strong reference to the referent. This may cause
+     * the garbage collector to treat it as strongly reachable until some later
+     * collection cycle.  The {@link #refersTo(Object) refersTo} method can be
+     * used to avoid such strengthening when testing whether some object is
+     * the referent of a reference object; that is, use {@code ref.refersTo(obj)}
+     * rather than {@code ref.get() == obj}.
+     *
      * @return   The object to which this reference refers, or
      *           {@code null} if this reference object has been cleared
+     * @see refersTo
      */
     @IntrinsicCandidate
     public T get() {
         return this.referent;
     }
 
+    /**
+     * Tests if this reference object refers to {@code obj}.  If {@code obj} is
+     * {@code null}, this method returns {@code true} if this reference object
+     * refers to {@code null} or has been cleared.
+     *
+     * @param   obj the object to compare with this reference object's
+     *          referent, or {@code null}.
+     * @return  {@code true} if and only if {@code obj} is the referent
+     *          of this reference object
+     * @since 16
+     */
+    public final boolean refersTo(T obj) {
+        return refersTo0(obj);
+    }
+
+    /* Implementation of refersTo() for non-phantom references.
+     */
+    native boolean refersTo0(Object o);
```

Comment:
1. Why is it named `refersTo0`? How about sth more descriptive, e.g.`refersToNative`?
2. Nit: including some text that this method is overridden in `PhantomReference.java` in the comments will be more informative and helpful; otherwise, one might be wondering how non-phantom and phantom refs are distinguished here.

Reply:
> See, for example, src/java.base/share/native/libjava/ConstantPool.c.

Got it; thank you.

> I'm not sure what else besides an override one might wonder about?

I don't know; maybe override is the only sensible choice. I just believe saying it explicitly is better than implicitly.

Reply:
OK.

Reply:
Thank you for the update.